nemo_skills/prompt/utils.py

```diff
@@ -239,15 +239,19 @@ def fill(
 
         if self.config.template:
             if multi_turn_key is None:
-                format_string = self.SYSTEM_FORMAT + self.TURN_BEGIN_FORMAT + "{generation}"
-                prompt_string = format_string.format(
-                    system=self.config.system,
-                    user=self.build_user_message(input_dict),
-                    generation=generation,
-                    **asdict(self.config.template),
+                prompt_string = self.SYSTEM_FORMAT.format(
```

Comment:
I'm confused by these changes - can you please explain why we need that?

Reply:
I just made the prompt_string construction logic similar to multi_turn. 
I think this flow should be easier to understand.
First, add the system part, then the user part, and, if the generation string is non-empty, the assistant part.
This version also adds the `.format()` to the system message which takes care of the double curly braces in the yaml string - {{}}.
Another difference from the prior version is that the assistant turn has an end token which we were missing earlier because the `format_string ` was ending on `{generation}`.